scripts/tests/test_zkai_native_seq32_attention_mlp_generated_proof_object_builder_gate.py

```
@@ -0,0 +1,305 @@
+        original = self.gate.inventory_gate.source_gate.atomic_write_text
+        calls = 0
+        with tempfile.TemporaryDirectory(dir=self.gate.EVIDENCE_DIR) as tmpdir:
+            tmp = pathlib.Path(tmpdir)
+            json_out = tmp / "builder.json"
+            tsv_out = tmp / "builder.tsv"
+
+            def fail_fourth_write(path: pathlib.Path, text: str) -> None:
+                nonlocal calls
+                calls += 1
+                if calls == 4:
+                    raise OSError("second final publish failed")
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)